terraform/test-fixtures/plan-count-inc-index-list/main.tf

```diff
@@ -0,0 +1,24 @@
+variable "count" {
+    default = 2
+}
+
+resource "aws_instance" "foo" {
+    count = "${var.count}"
+    foo = "foo"
+}
+
+resource "aws_instance" "bar" {
+    count = "${var.count}"
+    foo = "${element(aws_instance.foo.*.foo,count.index)}-bar"
```

Comment:
Interestingly, even though this resource is using the same element-splat-count-index technique as `aws_volume_attachment`, it is not marked for re-creation during the plan stage.